niet/actions.py

```diff
@@ -0,0 +1,36 @@
+def delete(input_dict, search):
+
+    if type(search) == list:
+        for path in search:
+            input_dict = delete(input_dict, path)
+    else:
+
+        # Remove whole dict
+        if search == ".":
+            input_dict = {}
+
+        # String begins with a dot
+        if search.startswith("."):
+            search = search[1:]
+
+        # get cur_search and next
+        if "." in search:
+            cur_search = search[0:(search.index("."))]
+            next_search = search[(search.index(".") + 1):]
+        else:
+            cur_search = search
+            next_search = ""
```

Comment:
I continue to think that we need to extract this part out of here (in a separate and dedicated function).

It can be called there and the function can return the both vars `cur_search` and `next_search` and they also can be initialized like this:
```python
cur_search, next_search = searching_path(seach)
```

Where `searching_path` will be your extract.

It can allow to use to more safely upgrade this part in the future by surrounding it with some dedicated unit tests. In the future we could by example introduce the list index etc...